cmd/nats-blackbox-exporter/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/snapp-incubator/nats-blackbox-exporter/internal/config"
+	"github.com/snapp-incubator/nats-blackbox-exporter/internal/logger"
+	"go.uber.org/zap"
+)
+
+var publishInterval = 2 * time.Second
+
+func main() {
+	cfg := config.New()
+
+	logger := logger.New(cfg.Logger)
+
+	nc, err := nats.Connect(cfg.NATS.URL)
+	if err != nil {
+		logger.Fatal("nats connection failed", zap.Error(err))
+	}
+
+	logger.Info("nats connection successful",
+		zap.String("connected-addr", nc.ConnectedAddr()),
+		zap.Strings("discovered-servers", nc.DiscoveredServers()))
+
+	nc.SetDisconnectErrHandler(func(_ *nats.Conn, err error) {
+		logger.Fatal("nats disconnected", zap.Error(err))
+	})
+
+	nc.SetReconnectHandler(func(_ *nats.Conn) {
+		logger.Warn("nats reconnected")
+	})
+
+	go subscribe(nc, logger)
+
+	go publish(nc, logger)
+
+	waitForSignal(logger)
+}
+
+func publish(nc *nats.Conn, logger *zap.Logger) {
+	for {
+		err := nc.Publish("subject1", []byte("Hello, NATS!"))
```

Comment:
subjects and target stream should be configurable ( something like example config mentioned in confluence doc )

Reply:
I put the subject as an input to my subscribe and publish functions. Do you think that's enough for now?

Reply:
yeah the subject it self is fine. but later you can make it configurable. also it has less overhead if you do it in this stage

Reply:
I added DefaultSubject to the configuration. If Subscribe and Publish are called with an empty string as the subject, this default value will be used.